src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -259,12 +259,47 @@ export default function SavedObjectFactory(es, kbnIndex, Promise, Private, Notif
     }
 
     /**
+     * Attempts to create the current object using the serialized source. If an object already
+     * exists, a warning message requests an overwrite confirmation.
+     * @param source - serialized version of this object (return value from this.serialize())
+     * What will be indexed into elasticsearch.
+     * @returns {Promise} - A promise that is resolved with the objects id if the object is
+     * successfully indexed. If the overwrite confirmation was rejected, an error is thrown with
+     * a confirmRejected = true parameter so that case can be handled differently than
+     * a create or index error.
+     * @resolved {String} - The id of the doc
+     */
+    const createSource = (source) => {
+      return docSource.doCreate(source)
+        .catch((err) => {
+          // record exists, confirm overwriting
+          if (_.get(err, 'origError.status') === 409) {
+            const confirmMessage = 'Are you sure you want to overwrite ' + this.title + '?';
```

Comment:
The import is matching on the id, not the title.

Good catch on the error message. What if it said `Are you sure you want to overwrite the object with id 50416250-cdf6-11e6-8394-6511ff44915a and replace it with Newest Dashboard?`?

Pretty ugly and I hate exposing the id. :(  I suppose I could change the logic and make a get query to es before the doCreate and check for document existence that way, but that might be overkill.

For what it's worth, this preserves the old logic which did the same exact thing, so at least we are no worse off than we were before.  I'm leaning towards leaving it as is and filing an issue to improve the error message as a separate issue.  This is at least better than no overwrite warning.